app/models/flight_m/copy_to_category.rb

```diff
@@ -0,0 +1,34 @@
+module FlightM
+  module CopyToCategory
+    def copy_to_category(category)
+      new_attrs = self.attributes
+      new_attrs.delete('id')
+      new_attrs.delete('category_id')
+      new_attrs['category_id'] = category.id
+      flight = Flight.find_by(contest_id: self.contest_id,
+        name: self.name, category_id: category.id)
+      if (flight)
+        flight.update!(new_attrs)
+      else
+        new_attrs['category_id'] = category.id
+        flight = Flight.create!(new_attrs)
+      end
+      copy_pilot_flights_to(flight)
+      flight
+    end
+
+    def copy_pilot_flights_to(flight)
+      new_pilots = self.pilot_flights.collect(&:pilot_id)
+      prior_pilots = flight.pilot_flights.collect(&:pilot_id)
+      flight.pilot_flights.where(
```

Comment:
It took me a while to deduce the purpose of this loop and I'm still not 100% certain. You want `flight.pilot_flights` to have a copy of the data in `self.pilot_flights`, but need to account for the possibility that this same Flight was already copied to the synthetic category? So you delete any `flight.pilot_flight` records that don't appear in `self.pilot_flights`. Yes, no, maybe?

You could replace lines 21-27 with:
    self.pilot_flights.delete_all # or destroy_all if there are relational dependencies
Then the loop in lines 28-31 does all the copying you need.